cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java

```diff
@@ -150,31 +151,42 @@ public void testInvalidAppWithDuplicateDatasets() throws Exception {
     AppRequest<AppWithDuplicateData.ConfigClass> createRequest = new AppRequest<>(
       new ArtifactSummary(artifactId.getName(), artifactId.getVersion().getVersion()),
       new AppWithDuplicateData.ConfigClass(false, false, false));
-    deployApplication(appId, createRequest);
+    try {
+      deployApplication(appId, createRequest);
+    } finally {
+      deleteApplication(appId);
```

Comment:
currently, the only way of doing that would be to perform a delete again, we don't have a ``getApplication`` API in ``TestBase``. 

Do you think we should either add a ``getApplication`` in ``TestBase`` or an ``exists`` API in ``ApplicationManager``? Or just verifying using another ``deleteApplication`` call is fine at this stage, and implement the ``getApplication`` API when we have a more concrete use case for it?